core/shared/src/main/scala/org/http4s/Message.scala

```diff
@@ -176,6 +176,12 @@ sealed trait Message[F[_]] extends Media[F] { self =>
   def putHeaders(headers: Header.ToRaw*): Self =
     transformHeaders(_.put(headers: _*))
 
+  def putHeader(header: Header.Raw): Self =
+    transformHeaders(_.putOne(header))
```

Comment:
I've added this interface, which is not used currently, so that middleware like CORS (https://github.com/http4s/http4s/blob/8efa92281ea90e625b9b767dd06052277e78c818/server/shared/src/main/scala/org/http4s/server/middleware/CORS.scala#L223) can be updated to be more efficient via `putHeader`

Reply:
I understand that we want this to use `HeaderRaw` for a reason (we want the `.name`?), but it seems a bit off that `putHeaders` works on `Header.ToRaw` while `putHeader` works on `Header.Raw`. Maybe `putRawHeader` would be a more appropriate name, if we must have this?

But maybe we wouldn't need it, if `name` was made available on all `Header.ToRaw` in a more efficient way?

Reply:
Honestly I'm not entirely sure why `Header.ToRaw` exists. It's just a `List[Header.Raw]`. What we actually need, and all that `Headers` contains is `List[Header.Raw]`, so I don't really see why we need `ToRaw`.

Reply:
I've tried to answer in the other comment, `Header.ToRaw` exists to support an api that works smoothly with typed headers, key-value pairs, Header.Raw, foldable of Header.Raw, etc.
The typed header is especially relevant since it lets you do `putHeaders(ContentType(...), Accept(..))`, which is the same api that existed in the old Scala 2 encoding. In turn, typed headers correctly handle whether they two occurrences of the same header should replace each other, merged, or accumulated in a List

Reply:
in this particular case, you basically aren't allowing users to work smoothly with typed headers